geopandas/io/parquet.py

```diff
@@ -181,6 +181,45 @@ def _validate_metadata(metadata):
             raise ValueError("Only WKB geometry encoding is supported")
 
 
+def _geopandas_to_arrow(df, index=None):
+    """
+    Helper function with main, shared logic for to_parquet/to_feather.
+    """
+    from pyarrow import Table
+
+    warnings.warn(
+        "this is an initial implementation of Parquet file support and "
```

Comment:
"Parquet" doesn't fit here, and a more generic Arrow-oriented message may not be as obvious to users (since conversion to Arrow table is an implementation detail).  

The warning could be moved to the `_to_parquet` and `_to_feather` level to keep the warning specific to the output format.

Reply:
For now I just replaced "Parquet" in the warning an error messages with "Parquet/Feather". 
In principle we could template all those messages to fill in one of the two, as appropriate.